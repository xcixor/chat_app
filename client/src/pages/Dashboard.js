/* eslint-disable no-unused-vars */
import React from "react";
import PropTypes from "prop-types";
import { Container } from "react-bootstrap";
import { ChatArea } from "../features";

const messages = [
	{
		sender: "User 1",
		id: "adbcd-3cert-rgt",
		message: "Hi",
	},
	{
		sender: "User 2",
		id: "adbcd-23ert-rgt",
		message: "Hello, how are you doing",
	},
	{
		sender: "User 1",
		id: "adbcd-3cert-rgt",
		message: "Im fine, thank you.",
	},
	{
		sender: "User 2",
		id: "adbcd-23ert-rgt",
		message: "Good to hear",
	},
	{
		sender: "User 3",
		id: "adbcd-3abvt-rgt",
		message: "Hi",
	},
];
export default function Dashboard({ id }) {
	return (
		<Container style={{ height: "100vh" }} data-testid="chat-room">
			<ChatArea messages={messages} currentUser={id} />
		</Container>
	);
}

Dashboard.propTypes = {
	id: PropTypes.string.isRequired,
};
