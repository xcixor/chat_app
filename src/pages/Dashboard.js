/* eslint-disable no-unused-vars */
import React, { useState, useRef, useEffect } from "react";
import PropTypes from "prop-types";
import { Container, Row } from "react-bootstrap";
import { ChatArea, MessageForm } from "../features";
import useLocalStorage from "../hooks/useLocalStorage";

export default function Dashboard({ id }) {
	const [messages, setMessages] = useLocalStorage("messages", []);
	const [isConnectionOpen, setConnectionOpen] = useState(false);
	const [messageBody, setMessageBody] = useState("");
	const wss = useRef();
	const sendMessage = () => {
		if (messageBody) {
			console.log(messageBody);
			wss.current.send(
				JSON.stringify({
					sender: id,
					body: messageBody,
				})
			);
			setMessageBody("");
		}
	};
	useEffect(() => {
		wss.current = new WebSocket("ws://localhost:9500");
		wss.current.onopen = () => {
			console.log("Connection opened");
			setConnectionOpen(true);
		};
		wss.current.onmessage = (event) => {
			const data = JSON.parse(event.data);
			console.log(data, "data");
			setMessages((_messages) => [..._messages, data]);
		};
		return () => {
			console.log("Closing...");
			wss.current.close();
		};
	}, []);
	const scrollTarget = useRef(null);

	useEffect(() => {
		if (scrollTarget.current) {
			const offsetBottom =
				scrollTarget.current.offsetTop + scrollTarget.current.offsetHeight;
			window.scrollTo({ top: offsetBottom });
		}
	}, [messages]);
	return (
		<Container
			style={{ height: "calc(100% - 60px)", paddingBottom:"65px" }}
			data-testid="chat-room"
			ref={scrollTarget}
		>
			<ChatArea messages={messages} currentUser={id} />
			<Row
				style={{
					position: "fixed",
					bottom: "0",
					height: "60px",
					width: "100%",
				}}
			>
				<MessageForm
					onMessageSubmit={setMessageBody}
					messageBody={messageBody}
					sendMessage={sendMessage}
					isConnectionOpen={isConnectionOpen}
				/>
			</Row>
		</Container>
	);
}

Dashboard.propTypes = {
	id: PropTypes.string.isRequired,
};
