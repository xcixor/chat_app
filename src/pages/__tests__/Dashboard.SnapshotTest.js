import React from "react";
import Dashboard from "../Dashboard";
import renderer from "react-test-renderer";

it("Renders correctly", () => {
	const tree = renderer.create(
		<Dashboard id="random-string" messages={[]} setMessages={jest.fn()} />
	);

	expect(tree).toMatchSnapshot();
});
