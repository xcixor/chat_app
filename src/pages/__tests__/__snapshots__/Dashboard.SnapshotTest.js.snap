// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="container"
  data-testid="chat-room"
  style={
    Object {
      "height": "calc(100% - 100px)",
      "paddingBottom": "250px",
    }
  }
>
  <div
    className="chat-area container"
    style={
      Object {
        "backgroundColor": "#F0F0F0",
      }
    }
  />
  <div
    style={
      Object {
        "bottom": "0",
        "height": "60px",
        "left": "0",
        "position": "fixed",
        "right": "0",
        "width": "100%",
      }
    }
  >
    <div
      className="row"
    >
      <form
        className="d-flex justify-content-center align-items-stretch"
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Say something"
          required={true}
          type="text"
          value=""
        />
        <button
          className="p-3 btn btn-primary"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "width": "20%",
            }
          }
          type="submit"
        >
          Send
        </button>
      </form>
    </div>
  </div>
  <div />
</div>
`;
